amethyst_network/src/error.rs

```diff
@@ -0,0 +1,85 @@
+//! Module containing error handling logic.
+
+use std::{
+    error::Error as StdError,
+    fmt::{Display, Formatter, Result as FmtResult},
+    io,
+    sync::mpsc,
+};
+
+use crate::server::ServerSocketEvent;
+
+/// The `amethyst_network` result type.
+pub type NetworkError<T> = Result<T, NetworkErrorKind>;
+
+/// Wrapper for all errors who could occur in `amethyst_network`.
+#[derive(Debug)]
+pub enum NetworkErrorKind {
+    /// Error that could occur on the UDP-socket
+    UdpError(laminar::error::NetworkError),
+    /// Error that could occur whit IO.
+    IoError(io::Error),
+    /// Error that could occur when serializing whit `bincode`
+    SerializeError(bincode::ErrorKind),
+    /// Error that could occur when sending an `ServerSocketEvent` to some channel.
+    ChannelSendError(mpsc::SendError<ServerSocketEvent>),
+}
+
+impl StdError for NetworkErrorKind {
```

Comment:
honestly, I here would do `std::error::Error` I think that is better than renaming.

Reply:
Oke